### PR TITLE
Cleanup BCLDebug

### PR DESCRIPTION
**Pull request by jkotas**
- Delete correctness, performance and safe handle stacktrace logging
- Change remaining BCLDebug.Assert uses to Debug.Assert

### CODE REVIEW

**Comment by jkotas** [comment]
Contributes to #11389

File: src/mscorlib/src/Microsoft/Win32/RegistryKey.cs [hunk 1]
```diff
@@ -539,7 +539,7 @@ internal Object InternalGetValue(String name, Object defaultValue, bool doNotExp
             if (datasize < 0)
             {
                 // unexpected code path
-                BCLDebug.Assert(false, "[InternalGetValue] RegQueryValue returned ERROR_SUCCESS but gave a negative datasize");
+                Debug.Assert(false, "[InternalGetValue] RegQueryValue returned ERROR_SUCCESS but gave a negative datasize");
```
Comment (justinvp) [inline src/mscorlib/src/Microsoft/Win32/RegistryKey.cs]: Nit: `Debug.Fail`

File: src/mscorlib/src/Microsoft/Win32/OAVariantLib.cs [hunk 2]
```diff
@@ -101,7 +101,7 @@ private static int GetCVTypeFromClass(Type ctype)
         {
             Debug.Assert(ctype != null);
 #if _DEBUG
```
Comment (stephentoub) [inline src/mscorlib/src/Microsoft/Win32/OAVariantLib.cs]: Is this #if _DEBUG needed?  When is that set?

File: src/mscorlib/src/System/RtType.cs [hunk 3]
```diff
@@ -389,7 +389,7 @@ private unsafe T[] GetListByName(char* pName, int cNameLen, byte* pUtf8Name, int
                             list = PopulateInterfaces(filter);
                             break;
                         default:
-                            BCLDebug.Assert(false, "Invalid CacheType");
+                            Debug.Assert(false, "Invalid CacheType");
```
Comment (stephentoub) [inline src/mscorlib/src/System/RtType.cs]: Debug.Fail

File: src/mscorlib/src/System/Runtime/InteropServices/SafeHandle.cs [hunk 4]
```diff
@@ -179,7 +167,7 @@ protected SafeHandle(IntPtr invalidHandleValue, bool ownsHandle)
         // Migrating InheritanceDemands requires this default ctor, so we can mark it critical
         protected SafeHandle()
         {
-            BCLDebug.Assert(false, "SafeHandle's protected default ctor should never be used!");
+            Debug.Assert(false, "SafeHandle's protected default ctor should never be used!");
```
Comment (stephentoub) [inline src/mscorlib/src/System/Runtime/InteropServices/SafeHandle.cs]: Search/replace ```Debug.Assert(false, ``` => ```Debug.Fail(```

**Comment by jkotas** [comment]
@dotnet-bot test CentOS7.1 x64 Release Priority 1 Build and Test  please
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test please

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
Will create new PR - the CI is stuck with old job definition

**Pull request closed without merging**